tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/PoPTests.NetFwk.cs

```diff
@@ -541,6 +541,98 @@ public async Task InMemoryCryptoProvider_AlgIsPS256()
             Assert.IsTrue(canonicalJwk.Contains(@"""alg"":""PS256"""), "The canonical JWK should include the alg field with value PS256");
         }
 
+        [TestMethod]
+        public async Task InMemoryCryptoProvider_IntegrationTest()
+        {
+            // Arrange - create a Confidential Client Application with PoP configuration
+            var settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                .Create(settings.ClientId)
+                .WithExperimentalFeatures()
+                .WithAuthority(settings.Authority)
+                .WithClientSecret(settings.GetSecret())
+                .Build();
+
+            // Create a new InMemoryCryptoProvider and get its JWK
+            var cryptoProvider = new InMemoryCryptoProvider();
+            var canonicalJwk = cryptoProvider.CannonicalPublicKeyJwk;
+            var base64EncodedJwk = Base64UrlHelpers.Encode(Encoding.UTF8.GetBytes(canonicalJwk));
+
+            // Calculate the expected kid
+            string expectedKid;
+            using (var sha256 = SHA256.Create())
+            {
+                var kidBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(canonicalJwk));
+                expectedKid = Base64UrlHelpers.Encode(kidBytes);
+            }
+
+            // Use InMemoryCryptoProvider
+            var popConfig = new PoPAuthenticationConfiguration(new Uri(ProtectedUrl))
+            {
+                PopCryptoProvider = cryptoProvider,
```

Comment:
I don't think you need this, you can rely on the in-memory one

Reply:
Oh, I guess you defined one to be able to validate some stuff,

Reply:
The e2e does not work and the test is failing with the graph endpoint. Looks like and email thread has been started already to unblock

Reply:
can you add me to that email thread?

Reply:
it was a chat let me forward it to you